src/mds/CInode.cc

```diff
@@ -5577,4 +5577,9 @@ void CInode::get_subtree_dirfrags(std::vector<CDir*>& v) const
   }
 }
 
+bool CInode::only_pinned_by_subtrees() const
+{
+  return get_num_ref() > num_subtree_roots;
+}
```

Comment:
if only pinned by subtree roots, get_num_ref() should be equal to num_subtree_roots;

I prefer to add CInode::get_num_subree_roots() and check  get_num_ref()  == get_num_subree_roots() in MDCache.cc

Reply:
@ukernel sounds good, I will make the changes.

Reply:
https://tracker.ceph.com/issues/46727

Reply:
This should just go in the header file.